XboxKeyboardMouse/XboxController/Emulation/XboxStream.cs

```diff
@@ -31,46 +28,52 @@ public struct RECT {
 
         public static Thread tMouseMovement;
 
-        public static void ToggleCursor() {
+        public static void XboxAppDetector() {
             const String XBOXAPP = "Xbox";
             bool started = false;
 
             const int count = 512;
             StringBuilder text = new StringBuilder(count);
-            while (true) {
-                try { Thread.Sleep(1000); } 
-
-                // Ensures that the current thread does not
-                // just crash and cause issues
-                catch (ThreadAbortException) { break; }
-                
-
-                IntPtr handle = GetForegroundWindow();
-
-                if (GetWindowText(handle, text, count) > 0) {
-                    if (text.ToString().Equals(XBOXAPP) == false) {
-                        ShowAndFreeCursor();
-                        started = false;
-                        Program.MainForm.StatusWaiting();
-
-                        continue;
-                    }
-
-                    /*if (IsFullscreen(handle) == false) {
-                        ShowAndFreeCursor();
-                        started = false;
-                        Program.mainform.StatusWaiting();
-
-                        continue;
-                    }*/
-
-                    if (started == false) {
-                        LockAndHideCursor();
-                        started = true;
-                        Program.MainForm.StatusRunning();
+            try
```

Comment:
It looks like try / catch was initially for catching an aborted sleep.  I don't know why you moved it out?  We want to catch as close to the error as possible and the other lines of code don't ThreadAbortException as far as I know.

Reply:
Technically we shouldn't be relying on thread .Abort to begin with, and instead aim for a graceful shutdown (e.g. the shuttingDown flags as an indicator for exiting).  However, I'm pretty sure ThreadAbortException will be raised from wherever the thread happened to be when it aborts; Thread.Sleep is where the loop spends the vast majority of it's time, so it is unlikely to have casually observed it occurring anywhere but the sleep.  So this would theoretically be more reliable than wrapping just the sleep alone.  If unconvinced, one can experiment by removing the sleep (letting the CPU peg) and seeing if the thread can still abort and catch a ThreadAbortException.

Reply:
Ok sounds good